packages/nimble-components/src/table/specs/table-column-specs/table-column-elapsed-time-hld.md

```diff
@@ -0,0 +1,177 @@
+}
+... // do similar work for minutes and seconds
+```
+
+Other implementation details will conform to existing patterns all other table columns handle, such as `CellView` and `GroupHeaderView` implementations.
+
+## Alternative Implementations / Designs
+
+### Limited configuration
+
+_*Props/Attrs*_
+
+-   `field-name`: string
+-   `style`: `"short" | "narrow" | "long"`
+
+This would offer clients to display elapsed time in other forms, but not to intermix formats between the various parts.
+
+Examples:
+
+-   "short": "1 hr, 30 min, 2 sec"
+-   "narrow": "1h, 30m, 2s"
+-   "long": "1 hour, 30 minutes, 20 seconds"
+
+Pros:
+
+-   Offers a little more configuration
+-   Not much extra cost to provide more display flexibility
+-   Aligns with the `Intl.DurationFormat` API that is in development
+
+Cons:
+
+-   No immediate need identified that would leverage the flexibility
+-   Risk of introducing API that we ultimately would prefer not to use?
+
+### Full configuration
+
+This option would be more akin to the [`TableColumnDateText` API](https://github.com/ni/nimble/blob/990e415d40fa3abe36414c1736b2133f7cd0cd03/packages/nimble-components/src/table-column/date-text/index.ts#L57C2-L57C2), where we would mirror the API surface of the [`options` constructor parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat/DurationFormat#parameters) of the `Intl.DurationFormat` object.
+
+Recommendation with this approach (if we decided to use this initially) would be to not support the `"digital"` `style` mode.
+
+Pros:
+
+-   Offers greatest set of flexibility for clients including the ability to display 'days' and 'years'.
+
+Cons:
+
+-   Increases implementation cost
+-   Higher risk of API ultimately not aligning with what we eventually want (particularly if `Intl.DurationFormat` is never formally adopted).
+
+Neither of the above options seem critical to implement initially, as we can always add features as needed.
+
+### States
+
+N/A
+
+### Accessibility
+
+### Mobile
+
+N/A
+
+### Globalization
+
+#### Localization
+
+Localization will be enabled through the `lang` setting provided by the `nimble-theme-provider` or root `<html>` element, which will then be passed as the `locale` to the JS native APIs (e.g. `Intl.NumberFormat`).
+
+### Security
+
+N/A
+
+### Performance
+
+N/A
+
+### Dependencies
+
+None
+
+### Test Plan
+
+Standard unit testing. Test cases of interest:
+
+-   unusual number values (NaN, -∞, ∞, negative values)
+-   verifying locale update results in expected display update
+
+### Tooling
+
+N/A
+
+### Documentation
+
+This component will be documented via a new story in Storybook.
+
+---
+
+## Open Issues
+
+-   Should we expand set of displayed time units to include days and/or years? Without introducing an accompanying API (i.e. would show days and years as needed by default), this might result in a different display than what SLE currently results in.
```

Comment:
I personally don't see an issue showing 'days' as a unit larger than 'hours', but I think extending it up to 'years' may open a can of worms. With leap year, I'm not sure we can state that all years have the same number of seconds. Plus, I could see that by the time something has lasted a year (or multiple years), we might stop caring so much about the number of seconds for that particular value. I could see including years really being year+days rather than getting all the way down to something like `3 years, 269 days, 19 hours, 12 minutes, 36.86 seconds`.

Reply:
I also strongly support adding 'days' since SLE will likely have durations like test execution times which are greater than 24 hours. And that will somewhat alleviate the clunkiness of including commas as both thousands separators and unit separators (`5,000 hr, 2 min, 1 sec` becomes `208 days, 8 hr, 2 min, 1 sec`).

I'm ok with not supporting weeks/months/years for now due to unclear requirements and complexity about conversion factors.

And I don't think we need to add a configuration option to turn off days formatting initially. We can do that (and your other configuration alternatives) if a client asks for it.

Reply:
Added days, but no years.